src/sqlfluff/core/templaters/slicers/heuristic.py

```diff
@@ -54,90 +54,90 @@ def _slice_template(in_str: str, env: Environment) -> Iterator[RawFileSlice]:
         if elem_type == "data":
             yield RawFileSlice(raw, "literal", idx)
             idx += len(raw)
-            continue
-        str_buff += raw
-
-        if elem_type.endswith("_begin"):
-            # When a "begin" tag (whether block, comment, or data) uses
-            # whitespace stripping
-            # (https://jinja.palletsprojects.com/en/3.0.x/templates/#whitespace-control),
-            # the Jinja lex() function handles this by discarding adjacent
-            # whitespace from in_str. For more insight, see the tokeniter()
-            # function in this file:
-            # https://github.com/pallets/jinja/blob/main/src/jinja2/lexer.py
-            # We want to detect and correct for this in order to:
-            # - Correctly update "idx" (if this is wrong, that's a
-            #   potential DISASTER because lint fixes use this info to
-            #   update the source file, and incorrect values often result in
-            #   CORRUPTING the user's file so it's no longer valid SQL. :-O
-            # - Guarantee that the slices we return fully "cover" the
-            #   contents of in_str.
-            #
-            # We detect skipped characters by looking ahead in in_str for
-            # the token just returned from lex(). The token text will either
-            # be at the current 'idx' position (if whitespace stripping did
-            # not occur) OR it'll be farther along in in_str, but we're
-            # GUARANTEED that lex() only skips over WHITESPACE; nothing else.
-
-            # Find the token returned. Did lex() skip over any characters?
-            num_chars_skipped = in_str.index(raw, idx) - idx
-            if num_chars_skipped:
-                # Yes. It skipped over some characters. Compute a string
-                # containing the skipped characters.
-                skipped_str = in_str[idx : idx + num_chars_skipped]
-
-                # Sanity check: Verify that Jinja only skips over
-                # WHITESPACE, never anything else.
-                if not skipped_str.isspace():  # pragma: no cover
-                    templater_logger.warning(
-                        "Jinja lex() skipped non-whitespace: %s", skipped_str
+        else:  # pragma: no cover
```

Comment:
No meaningful changes here. I removed the `continue` above and moved this code into an `else` block so I could mark the whole section unused with a single `# pragma: no cover` comment.

I think this code could be removed, as `heuristic.py` is only used now in **one** case: When dbt compiles a template that does not make use of Jinja templating, it doesn't call Jinja. In this case, our approach of intercepting Jinja calls doesn't "work", because that call does not occur. Instead, we end up falling back to `heuristic.py`, but in this case, we're _guaranteed_ that the raw file does not use Jinja templating. That's why this block is now unused, because it handles the case where the raw file contains something other than literal text.